packages/fxa-auth-server/scripts/refund-unverified-accounts.ts

```diff
@@ -0,0 +1,445 @@
+    'trialing',
+    'unpaid',
+  ].includes(keyInput);
+};
+
+const parseDryRun = (dryRun: boolean | string) => {
+  return `${dryRun}`.toLowerCase() !== 'false';
+};
+
+const parseDateForFirestore = (date: number) => Math.floor(date / 1000);
+
+const parseStartDate = (date: string): number =>
+  parseDateForFirestore(
+    date ? Date.parse(date) : Date.parse('2022-12-23T00:00:00')
+  );
+
+const parseEndDate = (date: string): number =>
+  parseDateForFirestore(
+    date ? Date.parse(date) : Date.parse('2022-12-31T23:59:59')
+  );
+
+const parseSubscriptionStatus = (
+  status: string
+): Stripe.Subscription['status'] | undefined => {
+  if (status) {
+    if (!isOfStatus(status)) {
+      throw new Error(
+        `Please provide a valid status. Invalid status: ${status}`
+      );
+    } else {
+      return status as Stripe.Subscription['status'];
+    }
+  } else {
+    return undefined;
+  }
+};
+
+function getCollectionNames() {
+  const config = Container.get(AppConfig);
+  const prefix = `${config.authFirestore.prefix}stripe-`;
+  return {
+    customerCollection: `${prefix}customers`,
+    subscriptionCollection: `${prefix}subscriptions`,
+    invoiceCollection: `${prefix}invoices`,
+  };
+}
+
+const getUidFromPath = (doc: QueryDocumentSnapshot) => {
+  const { customerCollection, subscriptionCollection } = getCollectionNames();
+  const path = doc.ref.path;
+  const before = `${customerCollection}/`;
+  const after = `/${subscriptionCollection}`;
+  return path.substring(
+    path.indexOf(before) + before.length,
+    path.indexOf(after)
+  );
+};
+
+function convertToCsv(data: OutputData[] | ErrorOutput[]) {
+  const replacer = (key: string, value: any) => (value === null ? '' : value); // specify how you want to handle null values here
+  const header = Object.keys(data[0]);
+  return [
+    header.join(','), // header row first
+    ...data.map((row: OutputData | ErrorOutput) =>
+      header
+        .map((fieldName: string) =>
+          JSON.stringify(row[fieldName as keyof typeof row], replacer)
+        )
+        .join(',')
+    ),
+  ].join('\r\n');
+}
+
+async function writeCsv(data: OutputData[] | ErrorOutput[], file: string) {
+  if (!data.length) {
+    return;
+  }
+  const csv = convertToCsv(data);
+  try {
+    await fs.writeFile(`${file}_${parseDateForFirestore(Date.now())}.csv`, csv);
+  } catch (err) {
+    console.error(err);
+  }
+}
+
+async function readSkip(file: string) {
+  const data = await fs.readFile(file, { encoding: 'utf-8' });
+  return data.split(/\r?\n/);
+}
+
+/**
+ * Retrieve subscriptions by created date
+ */
+async function retrieveSubscriptionsByCreated(
+  startDate: number,
+  endDate: number,
+  status?: string
+) {
+  const firestore = Container.get(AuthFirestore);
+  const { subscriptionCollection } = getCollectionNames();
+
+  const baseQuery = firestore
+    .collectionGroup(subscriptionCollection)
+    .where('created', '>=', startDate)
+    .where('created', '<=', endDate);
```

Comment:
It's not possible to query the subscription collection in this manner at the moment. Are we planning on adding this field to a single-field index before running this so we can?